addons/mail/static/src/components/thread_view/thread_view_tests.js

```diff
@@ -1801,6 +1801,102 @@ QUnit.test('composer should be focused automatically after clicking on the send
     );
 });
 
+QUnit.test('Message in thread view should be pending first', async function (assert) {
+    assert.expect(2);
+
+    this.data['mail.channel'].records = [{
+        channel_type: 'channel',
+        id: 20,
+        is_pinned: true,
+        message_unread_counter: 0,
+        name: "General",
+    }];
+
+    await this.start({
+        async mockRPC(route, args) {
+            if (args.method === 'message_post') {
+                await new Promise(() => {});
+            }
+            return this._super(...arguments);
+        },
+    });
+
+    const thread = this.env.models['mail.thread'].findFromIdentifyingData({
+        id: 20,
+        model: 'mail.channel'
+    });
+    const threadViewer = this.env.models['mail.thread_viewer'].create({
+        hasThreadView: true,
+        thread: [['link', thread]],
+    });
+    await this.createThreadViewComponent(threadViewer.threadView, { hasComposer: true });
+
+    document.querySelector('.o_ComposerTextInput_textarea').focus();
+    await afterNextRender(() => document.execCommand('insertText', false, "hey !"));
+    await afterNextRender(() =>
+        document.querySelector('.o_Composer_buttonSend').click()
+    );
+    assert.containsOnce(
+        document.body,
+        '.o-pending',
+        "chatter should contain 1 pending message"
+    );
+
+    document.querySelector('.o_ComposerTextInput_textarea').focus();
+    await afterNextRender(() => document.execCommand('insertText', false, "hey !"));
+    await afterNextRender(() =>
+        document.querySelector('.o_Composer_buttonSend').click()
+    );
+    assert.containsN(
+        document.body,
+        '.o-pending',
+        2,
+        "chatter should contain 2 pending message"
+    );
+});
+
+    QUnit.test('Message in thread view should display a retry button when the queue is in error', async function (assert) {
```

Comment:
queue is technical, need to use functional terms, such as "message failed to send"